lib/utils/filters.js

```diff
@@ -53,39 +57,44 @@ const validateFilters = function(type, filterObj) {
  * @param {Object} filterObj A simple key-value pair object for setting filters.
  * @returns {String} Returns the query header if the filter is valid. Else, it will throw an error.
  */
-const filterBuilder = function(searchType, filterObj=null) {
-    if (filterObj) { validateFilters(searchType, filterObj); }
+const filterBuilder = function (searchType, filterObj = null) {
+	if (filterObj) {
+		validateFilters(searchType, filterObj);
+	}
 
-    let baseQuery = `query ($page: Int, $perPage: Int, $search: String) {
+	let baseQuery = `query ($page: Int, $perPage: Int, $search: String) {
         Page (page: $page, perPage: $perPage) { pageInfo { total currentPage lastPage hasNextPage perPage } `;
 
-    if (searchType === "activity") {
-        baseQuery = `query ($page: Int, $perPage: Int, $id: Int) {
+	if (searchType === "activity") {
+		baseQuery = `query ($page: Int, $perPage: Int, $id: Int) {
             Page (page: $page, perPage: $perPage) { pageInfo { total currentPage lastPage hasNextPage perPage } 
             activities (id: $id`;
-    }
-    if (searchType === "anime") {
-        baseQuery += "media (type: ANIME, search: $search";
-    }
-    if (searchType === "manga") {
-        baseQuery += "media (type: MANGA, search: $search";
-    }
-
-    if (filterObj) {
-        let filterKeys = Object.keys(filterObj);
-    
-        for (let x = 0; x < filterKeys.length; x++) {
-            let filterValue = filterObj[filterKeys[x]];
-
-            if (filterValue instanceof Array) {
-                filterValue = `[${filterValue.join(", ")}]`;
-            }
-
-            baseQuery += `, ${filterKeys[x]}: ${filterValue}`; 
-        }
-    }
-
-    return `${baseQuery}) {`;
+	} else if (searchType === "anime") {
+		baseQuery += "media (type: ANIME, search: $search";
+	} else if (searchType === "manga") {
+		baseQuery += "media (type: MANGA, search: $search";
+	}
+
+	if (filterObj) {
+		let filterKeys = Object.keys(filterObj);
+
+		for (let x = 0; x < filterKeys.length; x++) {
+			let filterValue = filterObj[filterKeys[x]];
+
+			// If a value is completely uppercase, we are going to assume it's an ENUM
+			if (filterValue instanceof Array) {
+				filterValue = `[${filterValue
+					.map((value) => value === value.toUpperCase() ? value : `"${value}"`)
+					.join(", ")}]`;
+			} else if (typeof filterValue === "string" && filterValue !== filterValue.toUpperCase()) {
+				filterValue = `"${filterValue}"`;
+			}
```

Comment:
I know I said it looked good but I think we can modify this into two if statements and not one if and one else if statement since we are checking for an Array instance or a string type.

Reply:
changing this to 2 if's causes the second statement to always be true, if the first statement is true, on the string checking, since the variable is always turned into a string in the first if

Reply:
Ah right right. Good catch. This is good then. 👍